frontend/components/ManageStakingPage/StakingContractSection/MigrateButton.tsx

```diff
@@ -101,75 +82,150 @@ export const MigrateButton = ({
     return validation.reason;
   }, [currentStakingContractInfo, validation]);
 
+  return {
+    popoverContent,
+    validation,
+    currentStakingContractId: selectedStakingProgramId,
+    setActiveStakingProgramId,
+  };
+};
+
+type MigrateButtonProps = {
+  stakingProgramId: StakingProgramId;
+};
+
+/**
+ * Contract migration button
+ */
+export const MigrateButton = ({
+  stakingProgramId: stakingProgramIdToMigrateTo,
+}: MigrateButtonProps) => {
+  const [isMigrating, setIsMigrating] = useState(false);
+
+  const { goto } = usePageState();
+  const {
+    setPaused: setIsServicePollingPaused,
+    selectedService,
+    selectedAgentType,
+    overrideSelectedServiceStatus,
+  } = useServices();
+
+  // balance
+  const { setIsPaused: setIsBalancePollingPaused } = useBalanceContext();
+
+  // modals
+  const { setMigrationModalOpen } = useModals();
+
+  // staking contract details, validation, popover content
+  const {
+    popoverContent,
+    validation,
+    setActiveStakingProgramId,
+    currentStakingContractId,
+  } = useMigrateButtonDetails(stakingProgramIdToMigrateTo);
+
+  const serviceConfigId = selectedService?.service_config_id;
+  const serviceTemplate = useMemo(
+    () =>
+      SERVICE_TEMPLATES.find(
+        (template) => template.agentType === selectedAgentType,
+      ),
+    [selectedAgentType],
+  );
+
+  const handleSwitchAndRunAgent = useCallback(async () => {
+    if (!serviceTemplate) return;
+    if (!serviceConfigId) return;
+
+    setIsServicePollingPaused(true);
+    setIsBalancePollingPaused(true);
+
+    const canUseMechMarketplace =
+      stakingProgramIdToMigrateTo === StakingProgramId.PearlBetaMechMarketplace;
```

Comment:
should better check by "mechType =  MechType.Marketplace" rather than by staking program Id
but again would clarify with Jose, MW should read on-chain if the activity checker is based on the mech marketplace